pkg/client/client.go

```diff
@@ -58,24 +57,36 @@ import (
 type ImmuClient interface {
 	Disconnect() error
 	IsConnected() bool
+	// Deprecated: grpc retry mechanism can be implemented with WithConnectParams dialOption
 	WaitForHealthCheck(ctx context.Context) (err error)
+	// Deprecated: grpc protocol is already handling
 	HealthCheck(ctx context.Context) error
```

Comment:
This could still be a valid operation - e.g. before some more complex operation I could want to ensure that the connection to the db is valid.

It wold also be useful for health checks (e.g. return unhealthy status of a whole service if it can't talk to the DB)

Reply:
Yes, it's enough to deprecate waitForHealthCheck